### PR TITLE
ES7: migrate compatible functions to ductile

### PR DESCRIPTION
**Pull request by ereteog**
> Close #threatgrid/iroh/issues/4216
> Related #threatgrid/iroh/issues/3072

This PR replaces clj-momo by ductile when it's compatible. I identified below all the functions that require a compatibility mode.
I recommend to activate "Hide whitespace changes" for `ctia.observable.routes` in which I fixed an indentation issue.

* migrated:
  * ctia.stores.es.crud ns
    * get-doc-with-index
    * handle-read 
    * handle-find
    * handle-query-string-count
    * handle-query-string-search:
   Proposed a compabitlity failover in ductile for the extraction of the total count which is different from ES5 and ES7. This count is used in particular for the pagination.
    * handle-aggregate
  * ctia.task.migration.store ns
     * store-size (document count)
     * sliced-queries (document query)
     * query-fetch-batch (document query)
     * rollover (index refresh)
     * batch-delete (index refresh / document delete-by-query)
     * purge-store
    * create-target-store!
    * finalize-migration!
  * ctia.entity.judgement.es-store
    * document search
  * ctia.store.es.schemas
    * ESConnState schema
  * ctia.stores.es.init
     * get-existing-indices (index get)
     * init-es-conn! (create index)
  * ctia.http.handler
     * exception handlers es-query-parsing-error
  * ctia.task.rollover
    * rollover-store: index rollover
  * ctia.task.settings
    * rollover-store: index rollover
  * ctia.task.check-es-stores
  * ctia.observable.routes
   paginate
  * ctia.stores.es.init-test
   index / conn
  * ctia.bundle.routes-test
   index open! / close!
  * ctia.task.rollover-test
   index / conn
  * ctia.task.settings-test
   index / conn
  * ctia.task.update-mapping-test
   index / conn

* not migrated:
  * ctia.stores.es.crud ns
    * handle create:
    the `_type` field is required in ES5 `_bulk` endpoint (handle-create enable to create multiple documents).
    * handle-update / handle-delete:
The ES7 call is not compatible with the ES7 one that requires `_doc` in the URI instead of the type.
{"error":{"root_cause":[{"type":"invalid_type_name_exception","reason":"Document mapping type name can't start with '_', found: [_doc]"}],"type":"invalid_type_name_exception","reason":"Document mapping type name can't start with '_', found: [_doc]"},"status":400}{"error":{"root_cause":[{"type":"invalid_type_name_exception","reason":"Document mapping type name can't start with '_', found: [_doc]"}],"type":"invalid_type_name_exception","reason":"Document mapping type name can't start with '_', found: [_doc]"},"status":400}
  * ctia.task.migration.store ns
    * get-migration / update-migration-store / store-migration:
    the get / update document APIs require a `_doc` path element in ES7 that is ES5 refuses.
    * create-target-store!
    the index pattern template is different for `ductile.index/create-template`
  * ctia.entity.identiy
    * create / get /delete
    incompatible URIs
  * ctia.stores.es.init:
    * update-settings! / update-mapping! / upsert-template! 
    incompatibility between ES5 and ES7 format (ES5 requires type)
  * ctia.http.handler
    * Remove clj-momo query parsing ns error when after
  * ctia.stores.es.mapping-test
    * bulk-create-doc
  * ctia.task.migration.migrate-es-stores-test
   get-doc / create-doc / delete-doc
  * ctia.task.migration.store-test
   get-doc
  * ctia.test-helpers.es
   get-doc

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

### CODE REVIEW

File: src/ctia/task/migration/store.clj [hunk 1]
```diff
@@ -1,11 +1,13 @@
 (ns ctia.task.migration.store
   (:require [clj-momo.lib.clj-time.coerce :as time-coerce]
             [clj-momo.lib.clj-time.core :as time]
-            [clj-momo.lib.es.conn :as conn]
+            [ductile.conn :as conn]
+            [ductile.document :as ductile.doc]
+            [ductile.index :as ductile.index]
```
Comment (frenchy64) [inline src/ctia/task/migration/store.clj:6]: Not a big deal, but I noticed these aliases are the same as the namespaces.

**Review by frenchy64 — APPROVED** [review]
Very nice!

File: src/ctia/stores/es/crud.clj [hunk 2]
```diff
@@ -364,20 +360,17 @@ It returns the documents with full hits meta data including the real index in wh
...
-      :as es-conn-state} :- ESConnState
-     {:keys [filter-map] :as search-query} :- SearchQuery
-     ident]
-    (let [query (make-search-query es-conn-state search-query ident)]
-      (d/count-docs conn
-                    index
-                    (name mapping)
-                    query))))
+(s/defn handle-query-string-count :- s/Int
+  "ES count handler using some mapping and schema"
```
Comment (vemv) [inline src/ctia/stores/es/crud.clj:364]: Looks like the docstring became outdated

File: src/ctia/stores/es/crud.clj [hunk 3]
```diff
@@ -422,21 +415,18 @@ It returns the documents with full hits meta data including the real index in wh
                                 :value (:doc_count %))
                     buckets)))
 
-(defn handle-aggregate
+(s/defn handle-aggregate
   "Generate an ES aggregation handler using some mapping and schema"
```
Comment (vemv) [inline src/ctia/stores/es/crud.clj:419]: Same

**Review by vemv — APPROVED** [review]
LGTM!

File: project.clj [hunk 4]
```diff
@@ -58,6 +58,7 @@
 
                  [threatgrid/ctim "1.0.20"]
                  [threatgrid/clj-momo "0.3.5"]
+                 [threatgrid/ductile "0.1.0-SNAPSHOT"]
```
[the remark this answers is not in the document]
Comment (ereteog) [reply project.clj:61]: Indeed I simply forgot to update it :-).
Here is the release: https://github.com/threatgrid/ductile/releases/tag/0.1.0

**Pull request merged**